sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/RestProxyUtils.java

```diff
@@ -220,4 +222,7 @@ public static HttpPipeline createDefaultPipeline() {
             .build();
     }
 
+    public static SwaggerInterfaceParser getSwaggerInterfaceParser(Class<?> swaggerInterface) {
+        return INTERFACE_PARSERS.computeIfAbsent(swaggerInterface, SwaggerInterfaceParser::new);
+    }
```

Comment:
I'd move this to `SwaggerInterfaceParser`. We can have two layers of caching there. I think it's better to keep them in one place.